mono/metadata/class-private-definition.h

```diff
@@ -130,6 +138,10 @@ struct _MonoClass {
 	MonoPropertyBag infrequent_data;
 };
 
+typedef enum {
+	MONOCLASS_BITFIELDS_HAS_REFERENCE = (1 << 16)
```

Comment:
This still looks like potential big-endian issue to me.

Reply:
I guess the same applies to this in this case:  https://github.com/mono/mono/blob/7a15b718735eccfac6ecd7467a09279d2f23e16e/mono/metadata/class-internals.h#L337 ?

Reply:
It does not. Firstly, `MonoVTableFlags` fits into one byte so it behaves identically on LE and BE machine. Secondly, it is not aliased over bitfields.

The issue is that the bit field layout over the `bitfields` int in the union may be byte-reversed. I haven't actually verified how the compiler on BE machine behaves but it's very likely it won't match correctly.

I am not sure if there's a nice way to solve it. You can do the anonymous union over single bytes of the bit field but it feels weird to read. Maybe a flag in `MonoVTableFlags` may be easier to read after all.

Reply:
Put the flag into MonoVTable.flags, its there for this purpose. It should be initialized in mono_class_create_runtime_vtable ().